modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Augmentation.java

```diff
@@ -552,4 +553,78 @@ public static String decodeBase64(String receiver) {
         // O(N) or faster depending on implementation
         return result.toArray(new String[0]);
     }
+
+    /**
+     * Access values in nested containers with a dot separated path.  Path elements are treated
+     * as strings for Maps and integers for Lists.  Throws 'IllegalArgumentException' if path does
+     * not exist. Throws 'NumberFormatException' if an path element for a List is not an integer.
+     */
+    private static Object getByPath(Object receiver, String path) {
+        Object value = getByPath(receiver, path, new missingValue());
```

Comment:
I think we can avoid having a special missingValue by passing in a Supplier as the default value to the primary methods, where here would would give an instance that throws an exception.

Reply:
Now using a Supplier.